src/core/client_channel/subchannel.cc

```diff
@@ -966,6 +1055,29 @@ bool Subchannel::PublishTransportLocked() {
     connected_subchannel_ = MakeRefCounted<NewConnectedSubchannel>(
         std::move(*call_destination), std::move(transport_destination), args_);
   }
+
+  // Extract security level from auth context
+  std::string security_level = "";
+  auto auth_context =
+      connecting_result_.channel_args.GetObjectRef<grpc_auth_context>();
+  if (auth_context != nullptr) {
+    grpc_auth_property_iterator it = grpc_auth_context_find_properties_by_name(
+        auth_context.get(), GRPC_TRANSPORT_SECURITY_LEVEL_PROPERTY_NAME);
+    const grpc_auth_property* prop = grpc_auth_property_iterator_next(&it);
+    if (prop != nullptr) {
+      std::string tsi_level(prop->value, strlen(prop->value));
```

Comment:
This should use `absl::string_view` instead of `std::string` to avoid an unnecessary string copy.